rspace/src/main/scala/coop/rchain/rspace/merger/instances/DiffStateMerger.scala

```diff
@@ -0,0 +1,317 @@
+package coop.rchain.rspace.merger.instances
+
+import cats.effect.Concurrent
+import cats.effect.concurrent.Ref
+import cats.implicits._
+import coop.rchain.rspace._
+import coop.rchain.rspace.channelStore.syntax.ConsumeMapping
+import coop.rchain.rspace.channelStore.{ChannelStore, DataJoinHash}
+import coop.rchain.rspace.history._
+import coop.rchain.rspace.internal.{toOrderedByteVectors, RichDatum, RichJoin, RichKont}
+import coop.rchain.rspace.merger.{computeChannelChange, ChannelChange, EventChain, StateMerger}
+import coop.rchain.rspace.syntax._
+import coop.rchain.rspace.trace.{COMM, Consume, Produce}
+import coop.rchain.shared.syntax._
+import coop.rchain.shared.{Log, Serialize, Stopwatch}
+import coop.rchain.store.LazyKeyValueCache
+
+/**
+  * State merger using diff of start and enf state to compute changes
+  * @param historyRepo repository for histories
+  * @param serializeC serizlized for channel (required for joins computations)
+  */
+final case class DiffStateMerger[F[_]: Concurrent: Log, C, P, A, K](
+    historyRepo: HistoryRepository[F, C, P, A, K],
+    serializeC: Serialize[C]
+) extends StateMerger[F] {
+
+  override def merge(mainState: History[F], toMerge: Seq[EventChain[F]]): F[History[F]] =
+    for {
+      // compute trie actions
+      trieActions <- Stopwatch.time(Log[F].debug(_))(
```

Comment:
@tgrospic I think this one can be another implementation for `Span`. what do you think?

Reply:
I see Span, Log, Metrics as a garbage inside the code. Doesn't really do anything, just makes code uglier. :)

But it's needed, so it should be minimally intrusive and just specify our intention. E.g. to measure time we just need to specify a block of code and unique identifier.

```scala
// We don't need to specify implementation
// It can be measured with Span and/or put inside the Log
_ <- Diag[F].measureTime(<unique-id>) {
  ...
}

// Or we just want to specify some event that can be configured
//  to put in the Log or Metrics endpoint
_ <- Diag[F].event(<unique-id>)
```
@zsluedem I'll create more elaborate info about that.